test/parallel/test-emit-after-uncaught-exception.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const async_hooks = require('async_hooks');
+
+const id_obj = {};
+let collect = true;
+
+const hook = async_hooks.createHook({
+  before(id) { if (collect) id_obj[id] = true; },
+  after(id) { delete id_obj[id]; },
+}).enable();
+
+process.once('uncaughtException', common.mustCall((er) => {
+  assert.strictEqual(er.message, 'bye');
+  collect = false;
+}));
+
```

Comment:
The first pass of the script is the "bootstrap", which always has an id of `1`. But as soon as it throws the stack is cleared and `async_uid_fields[kCurrentAsyncId] === 0` until another stack is entered. That answer your question?